bots-deployment-detector/src/agent.ts

```diff
@@ -1,43 +1,68 @@
+import { Finding, HandleTransaction, TransactionEvent, FindingSeverity, FindingType, ethers } from 'forta-agent';
+
 import {
-  BlockEvent,
-  Finding,
-  Initialize,
-  HandleBlock,
-  HandleTransaction,
-  HandleAlert,
-  AlertEvent,
-  TransactionEvent,
-  FindingSeverity,
-  FindingType,
-} from 'forta-agent';
-
-let findingsCount = 0;
-
-const handleTransaction: HandleTransaction = async (txEvent: TransactionEvent) => {
-  const findings: Finding[] = [];
-
-  return findings;
-};
+  nethermindDeployerAddress,
+  fortaContractAddress,
+  eventAgentUpdated,
+  eventAgentEnabled,
+  functionCreateAgent,
+  findingAgentInputs,
+  FindingAgentInput,
+} from './utils';
+
+export function provideHandleTransaction(deployerAddress: string, contractAddress: string): HandleTransaction {
+  return async function handleTransaction(txEvent: TransactionEvent) {
+    const findings: Finding[] = [];
+
+    if (txEvent.from.toLowerCase() !== deployerAddress.toLowerCase()) {
+      return findings;
+    }
+
+    const logs = txEvent.filterLog([eventAgentUpdated, eventAgentEnabled], contractAddress);
+
+    const fn = txEvent.filterFunction(functionCreateAgent, contractAddress);
+
+    logs.forEach((log) => {
+      const { agentId, by, chainIds } = log.args;
+      const { name } = log;
+
+      let findingInput: FindingAgentInput;
+
+      let metadataInput;
 
-// const initialize: Initialize = async () => {
-//   // do some initialization on startup e.g. fetch data
-// }
+      if (name === 'AgentUpdated') {
+        findingInput = fn.length > 0 ? findingAgentInputs.create : findingAgentInputs.update;
```

Comment:
the only important thing that needs to change. here, in order to emit a `createAgent` alert, you rely on the transaction having logs. for some reason, this `createAgent` could be used and there might be no events emitted. in those cases you would be missing the alert. so another `forEach` for `fn` is required.

Reply:
The createAgent function emits AgentUpdated event and updateAgent function also emits AgentUpdated. 
But if I write a forEach for fn also, then there might be duplicate alerts for the same thing. One from event logs and one from fn. 

So, to ensure that there will be no duplicate create Alerts, I am storing the metadata of fn logs and checking them against event logs metadata. See the below Image. I assume this should be fine. Please let me know if there is some other way I can do this.

![image](https://github.com/Macbeth98/forta-bots/assets/29257812/f758448e-367e-4c3d-bfaf-4dcf94362c9b)

Reply:
> The createAgent function emits AgentUpdated event and updateAgent function also emits AgentUpdated.

you're right about it. then why don't you use `updateAgent` function abi like you did for `createAgent`?

Reply:
Yes I can do that. I will change the code to reflect that.

Reply:
I just updated the Agent. I replaced `AgentUpdate` event with `updateAgent` function ABI.